iox_catalog/src/postgres.rs

```diff
@@ -1906,6 +1906,43 @@ LIMIT $3;
         .map_err(|e| Error::SqlxError { source: e })
     }
 
+    async fn partitions_with_small_l1_file_count(
+        &mut self,
+        shard_id: ShardId,
+        small_size_threshold_bytes: i64,
+        min_small_file_count: usize,
+        num_partitions: usize,
+    ) -> Result<Vec<PartitionParam>> {
+        // This query returns partitions with at least `min_small_file_count` small L1 files,
+        // where "small" means no bigger than `small_size_threshold_bytes`, limited to the top `num_partitions`.
+        sqlx::query_as::<_, PartitionParam>(
+            r#"
+SELECT parquet_file.partition_id, parquet_file.shard_id, parquet_file.namespace_id,
+       parquet_file.table_id,
+       COUNT(*) AS l1_file_count,
```

Comment:
```suggestion
       COUNT(1) AS l1_file_count,
```
We are not clear what Postgres does with `count(*)` but `count(1)` is a count on a constant value 1 which we think will do the minimum work